extensions/amp-next-page/0.2/service.js

```diff
@@ -176,23 +194,37 @@ export class NextPageService {
         page.relativePos === PageRelativePos.CONTAINS_VIEWPORT
       ) {
         if (!page.isVisible()) {
-          page.setVisible(VisibilityState.VISIBLE);
+          page.setVisibility(VisibilityState.VISIBLE);
         }
-        // Hide the previous page
-        const prevPage = this.pages_[index + this.scrollDirection_];
-        if (
-          prevPage &&
-          prevPage.relativePos === PageRelativePos.LEAVING_VIEWPORT &&
-          prevPage.isVisible()
-        ) {
-          prevPage.setVisible(VisibilityState.HIDDEN);
+        // Hide the previous pages
+        let prevPageIndex = index + this.scrollDirection_;
+        while (prevPageIndex >= 0 && prevPageIndex < this.pages_.length) {
+          const prevPage = this.pages_[prevPageIndex];
+          if (
+            prevPage &&
+            (prevPage.relativePos === PageRelativePos.LEAVING_VIEWPORT ||
+              prevPage.relativePos === PageRelativePos.OUTSIDE_VIEWPORT ||
+              prevPage === this.initialPage_) &&
+            prevPage.isVisible()
+          ) {
+            prevPage.setVisibility(VisibilityState.HIDDEN);
+          }
+          prevPageIndex += this.scrollDirection_;
         }
       } else if (page.relativePos === PageRelativePos.OUTSIDE_VIEWPORT) {
         if (page.isVisible()) {
-          page.setVisible(VisibilityState.VISIBLE);
+          page.setVisibility(VisibilityState.HIDDEN);
         }
       }
     });
+
+    // If no page is visible then the host page should be
+    if (
+      !this.pages_.some(page => page.isVisible()) &&
```

Comment:
If there are no pages that are visible, doesn't that imply the `initialPage` is also not visible? If so, do we need to also check that below? If not, why is that the case?

Reply:
That's correct! No need for the second check